src/index.js

```diff
@@ -157,13 +260,14 @@ function addMetadata({ types: t }) {
       /**
        * Do transforms for adding our test metadata expressions to beforeEach.
        * For each top-level module (in QUnit there can be sibling and/or nested modules), only 1 beforeEach should apply,
-       * and so only 1 beforeEach will be added/transformed per top-level module. As Babel traverses through each call
-       * expression, we're only concerned about operating on call expressions inside of top-level module calls.
+       * and so only 1 beforeEach will be added/transformed per top-level module.
        *
-       * While inside a top-level module, we check the current call expression for either of these conditions:
-       *   1. if it's a beforeEach, then we add our test metadata expressions to it.
-       *   2. if it's a call to either "test" or to a nested "module", then we want to add a new beforeEach just above it
-       * If the current call is neither of these, then do nothing. Babel will move on to the next call expression.
+       * While inside a top-level module, we:
+       * 1. check for any existing beforeEach call
+       * 2. else, check for any test setup call expressions, e.g. setupApplication(hooks)
+       *
+       * If there is an existing beforeEach, we add our statements to it. Otherwise, we add a new beforeEach, either
+       * after any test setup calls or at the top of the test module.
        *
        * The transformed beforeEach would look like:
```

Comment:
I worry this comment is going to age out, as evidenced by your having to keep updating it. I'd much prefer to keep the code with semantically named variables so they express the intent.

Reply:
Ok, I cleaned it up and only left the part about sibling/nested modules and how the beforeEach would look like.